src/main/kotlin/listener/ChatListener.kt

```diff
@@ -1,45 +1,45 @@
-package listener
+package chattore.listener
 
-import ChattORE
-import entity.ChattORESpec
-import formatGlobal
-import net.md_5.bungee.api.connection.ProxiedPlayer
-import net.md_5.bungee.api.event.ChatEvent
-import net.md_5.bungee.api.event.TabCompleteEvent
-import net.md_5.bungee.api.plugin.Listener
-import net.md_5.bungee.event.EventHandler
+import chattore.ChattORE
+import com.velocitypowered.api.event.Subscribe
+import com.velocitypowered.api.event.command.CommandExecuteEvent
+import com.velocitypowered.api.event.player.PlayerChatEvent
+import com.velocitypowered.api.event.player.TabCompleteEvent
+import com.velocitypowered.api.proxy.Player
+import chattore.entity.ChattORESpec
+import chattore.formatGlobal
 
 class ChatListener(
     private val chattORE: ChattORE
-) : Listener {
-    @EventHandler
+) {
+    @Subscribe
     fun onTabComplete(event: TabCompleteEvent) {
-        // Cancels propagated command suggestions from downstream servers
-        if (
-            event.cursor.matches(
-                Regex("""/(msg|message|vmsg|vmessage|whisper|tell)\s.*\s.*""")
-            ) // Cancels all direct message suggestions after a recipient is specified
-            || event.cursor.matches(
-                Regex("""/(me|helpop|ac|reply|r)\s.*""")
-            ) // Cancels all suggestions
-        ) {
-            event.isCancelled = true
+        // TODO: Autocomplete player names and stuff idk
+        event.suggestions.clear()
+    }
+
+    @Subscribe
+    fun onChatEvent(event: PlayerChatEvent) {
+        event.result = PlayerChatEvent.ChatResult.denied()
+
+        val pp = event.player
+        pp.currentServer.ifPresent { server ->
+            chattORE.broadcastChatMessage(server.serverInfo.name, pp.uniqueId, event.message)
         }
     }
-    @EventHandler
-    fun onChatEvent(event: ChatEvent) {
-        if (event.isCommand || event.isProxyCommand) {
+
+    @Subscribe
+    fun onCommandEvent(event: CommandExecuteEvent) {
+        val commandSource = event.commandSource
```

Comment:
Here you could do hax with
```suggestion
        val commandSource = event.commandSource as? Player ?: return
```
Then the if is not needed